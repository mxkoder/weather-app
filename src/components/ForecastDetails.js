import React from "react";
import PropTypes from "prop-types";
import {
  WiDirectionUp,
  WiDirectionDownLeft,
  WiDirectionDownRight,
  WiDirectionDown,
  WiDirectionLeft,
  WiDirectionRight,
  WiDirectionUpLeft,
  WiDirectionUpRight,
} from "react-icons/wi";
import "../styles/ForecastDetails.css";

const ForecastDetails = ({ forecast }) => {
  const { date, temperature, wind, humidity } = forecast;

  const formattedDate = new Date(date).toDateString();

  let windArrowDirection = <WiDirectionUp />;
  if (wind.direction === "sw" || wind.direction === "ssw") {
    windArrowDirection = <WiDirectionUpRight />;
  }
  if (wind.direction === "w" || wind.direction === "wsw") {
    windArrowDirection = <WiDirectionRight />;
  }
  if (wind.direction === "nw" || wind.direction === "wnw") {
    windArrowDirection = <WiDirectionDownRight />;
  }
  if (wind.direction === "n" || wind.direction === "nnw") {
    windArrowDirection = <WiDirectionDown />;
  }
  if (wind.direction === "ne" || wind.direction === "nne") {
    windArrowDirection = <WiDirectionDownLeft />;
  }
  if (wind.direction === "e" || wind.direction === "ene") {
    windArrowDirection = <WiDirectionLeft />;
  }
  if (wind.direction === "se" || wind.direction === "ese") {
    windArrowDirection = <WiDirectionUpLeft />;
  }

  return (
    <div className="forecast-details" data-testid="forecast-summary">
      <div className="forecast-details__date">{formattedDate}</div>
      <div className="forecast-details__max-temperature">
        Max Temperature: {temperature.max}&deg;C
      </div>
      <div className="forecast-details__min-temperature">
        Min Temperature: {temperature.min}&deg;C
      </div>
      <div className="forecast-details__humidity">
        Humidity: {humidity}&#37;
      </div>
      <div className="forecast-details__wind-speed">
        Wind Speed: {wind.speed} mph
      </div>
      <div className="forecast-details__wind-direction">
        Wind Direction: {windArrowDirection}
      </div>
    </div>
  );
};

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number.isRequired,
    temperature: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number,
    }).isRequired,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
    humidity: PropTypes.number.isRequired,
  }).isRequired,
};

export default ForecastDetails;
