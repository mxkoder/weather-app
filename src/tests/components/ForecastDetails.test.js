import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1525046400000,
    temperature: {
      min: 12,
      max: 22,
    },
    humidity: 30,
    wind: {
      speed: 40,
      direction: "s",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText } = render(<ForecastDetails forecast={validProps} />);

    expect(getByText("Mon Apr 30 2018")).toHaveClass("forecast-details__date");
    expect(getByText("Max Temperature: 22°C")).toHaveClass(
      "forecast-details__max-temperature"
    );
    expect(getByText("Min Temperature: 12°C")).toHaveClass(
      "forecast-details__min-temperature"
    );
    expect(getByText("Humidity: 30%")).toHaveClass(
      "forecast-details__humidity"
    );
    expect(getByText("Wind Speed: 40 mph")).toHaveClass(
      "forecast-details__wind-speed"
    );
  });
});
