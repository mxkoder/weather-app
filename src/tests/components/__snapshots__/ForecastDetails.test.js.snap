// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-summary"
  >
    <div
      class="forecast-details__date"
    >
      Mon Apr 30 2018
    </div>
    <div
      class="forecast-details__max-temperature"
    >
      Max Temperature: 22°C
    </div>
    <div
      class="forecast-details__min-temperature"
    >
      Min Temperature: 12°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity: 30%
    </div>
    <div
      class="forecast-details__wind-speed"
    >
      Wind Speed: 40 mph
    </div>
    <div
      class="forecast-details__wind-direction"
    >
      Wind Direction: 
      <svg
        fill="currentColor"
        height="1em"
        id="Layer_1"
        stroke="currentColor"
        stroke-width="0"
        version="1.1"
        viewBox="0 0 30 30"
        width="1em"
        x="0px"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <path
          d="M9.95,10.87c-0.01,0.35,0.1,0.65,0.34,0.9s0.53,0.37,0.89,0.36c0.34,0.02,0.63-0.1,0.88-0.37l1.66-1.64v10.3
	c-0.01,0.35,0.11,0.64,0.36,0.88s0.55,0.35,0.92,0.34c0.34,0.02,0.64-0.09,0.89-0.32s0.39-0.53,0.4-0.88v-10.3l1.64,1.64
	c0.23,0.24,0.53,0.37,0.88,0.37c0.36,0,0.66-0.12,0.9-0.36s0.36-0.53,0.36-0.89c-0.02-0.36-0.15-0.64-0.4-0.85l-3.74-3.84
	c-0.24-0.23-0.55-0.37-0.92-0.4c-0.37,0.02-0.68,0.16-0.92,0.41l-3.75,3.81C10.08,10.25,9.95,10.53,9.95,10.87z"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
